compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -7266,12 +7266,112 @@ TR::Register *OMR::X86::TreeEvaluator::vmushrEvaluator(TR::Node *node, TR::CodeG
 
 TR::Register *OMR::X86::TreeEvaluator::vrolEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 {
-    return TR::TreeEvaluator::vectorBinaryArithmeticEvaluator(node, cg);
+    TR::DataType type = node->getType();
+    TR::DataType et = type.getVectorElementType();
+    TR::VectorLength vl = type.getVectorLength();
+
+    TR_ASSERT_FATAL(et != TR::Int8 && !et.isFloatingPoint(),
+        "Vector rotate evaluator does not support 8-bit types nor floating-point types");
+
+    if (cg->comp()->target().cpu.supportsFeature(OMR_FEATURE_X86_AVX512F) && et != TR::Int16) {
```

Comment:
I made a comment earlier about caching the numerous `cpu` reaches in this method into a local, like this:
`TR::CPU *cpu = &cg->comp()->target().cpu;`